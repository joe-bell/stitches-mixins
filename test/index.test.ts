import { createStitches } from "@stitches/core";
import { mixins } from "stitches-mixins";

const theme = { colors: { orchid: "orchid" }, space: { md: "1rem" } };

const customMixins = {
  orchidShadow: {
    boxShadow: "0 25px 50px -12px $orchid",
  },
};

describe("mixins", () => {
  test("individual mixins can be included and overridden", () => {
    const stitches = createStitches({
      theme,
      utils: {
        include: mixins(),
      },
    });

    const component = stitches.css({
      include: "box",
      margin: "$md",
    });

    // CSS should contain theme etc…
    expect(stitches.toString()).toMatchSnapshot();
    // Create component CSS
    component.toString();
    // CSS should now contain component CSS
    expect(stitches.toString()).toMatchSnapshot();
  });

  test("multiple mixins can be included and overridden", () => {
    const stitches = createStitches({
      theme,
      utils: {
        include: mixins(),
      },
    });

    const component = stitches.css({
      include: ["box", "minHeightScreen"],
      margin: "$md",
    });

    // CSS should contain theme etc…
    expect(stitches.toString()).toMatchSnapshot();
    // Create component CSS
    component.toString();
    // CSS should now contain component CSS
    expect(stitches.toString()).toMatchSnapshot();
  });

  test("custom mixins can be included", () => {
    const stitches = createStitches({
      theme,
      utils: {
        include: mixins(customMixins),
      },
    });

    const component = stitches.css({
      include: ["box", "orchidShadow"],
    });

    // CSS should contain theme etc…
    expect(stitches.toString()).toMatchSnapshot();
    // Create component CSS
    component.toString();
    // CSS should now contain component CSS
    expect(stitches.toString()).toMatchSnapshot();
  });

  test("mixins can be nested", () => {
    const stitches = createStitches({
      theme,
      utils: {
        include: mixins(customMixins),
        someShorthandUtil: (value: boolean) =>
          value && {
            boxShadow: "0 25px 50px -12px orchid",
          },
      },
    });

    const component = stitches.css({
      include: ["box", "orchidShadow", "screenReaderOnly"],
      "&:focus": {
        include: "notScreenReaderOnly",
      },
    });

    // CSS should contain theme etc…
    expect(stitches.toString()).toMatchSnapshot();
    // Create component CSS
    component.toString();
    // CSS should now contain component CSS
    expect(stitches.toString()).toMatchSnapshot();
  });
});
