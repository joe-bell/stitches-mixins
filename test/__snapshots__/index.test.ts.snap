// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixins custom mixins can be included 1`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}"`;

exports[`mixins custom mixins can be included 2`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}--sxs{--sxs:2 c-jNRlAV}@media{.c-jNRlAV{min-width:0;box-sizing:border-box}.c-jNRlAV::before,.c-jNRlAV::after{box-sizing:inherit}}"`;

exports[`mixins individual mixins can be included and overridden 1`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}"`;

exports[`mixins individual mixins can be included and overridden 2`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}--sxs{--sxs:2 c-fovnzX}@media{.c-fovnzX{min-width:0;box-sizing:border-box}.c-fovnzX::before,.c-fovnzX::after{box-sizing:inherit}.c-fovnzX{margin:var(--space-md)}}"`;

exports[`mixins mixins can be nested 1`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}"`;

exports[`mixins mixins can be nested 2`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}--sxs{--sxs:2 c-jqIsIq}@media{.c-jqIsIq{min-width:0;box-sizing:border-box}.c-jqIsIq::before,.c-jqIsIq::after{box-sizing:inherit}.c-jqIsIq{box-shadow:0 25px 50px -12px var(--shadows-orchid);position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;border-width:0}.c-jqIsIq:focus{position:static;width:auto;height:auto;padding:0;margin:0;overflow:visible;clip:auto;white-space:normal}}"`;

exports[`mixins multiple mixins can be included and overridden 1`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}"`;

exports[`mixins multiple mixins can be included and overridden 2`] = `"--sxs{--sxs:0 t-gCfUPt}@media{:root,.t-gCfUPt{--colors-orchid:orchid;--space-md:1rem}}--sxs{--sxs:2 c-bIRQaM}@media{.c-bIRQaM{min-width:0;box-sizing:border-box}.c-bIRQaM::before,.c-bIRQaM::after{box-sizing:inherit}.c-bIRQaM{min-height:100vh}@supports (-webkit-touch-callout: none){.c-bIRQaM{min-height:-webkit-fill-available}}.c-bIRQaM{margin:var(--space-md)}}"`;
